File: 13-roman-to-integer/romanToInt.cpp
#include <iostream>
#include <string>
#include <map>
using namespace std;

int romanToInt(string s) {
    map<char, int> num;
    num['I'] = 1;
    num['V'] = 5;
    num['X'] = 10;
    num['L'] = 50;
    num['C'] = 100;
    num['D'] = 500;
    num['M'] = 1000;
    int result = num[s[s.length()-1]];
    for(int i=s.length()-2;i>=0;i--){
        if(num[s[i]] >= num[s[i+1]]){
            result += num[s[i]];
        }else {
            result -= num[s[i]];
        }
    }
    return result;
}

int main() {
    printf("%d\n", romanToInt("DCXXI"));
    return 0;
}

File: 013-roman-to-integer/romanToInt.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function (s) {
    var obj = {
        'I': 1,
        'V': 5,
        'X': 10,
        'L': 50,
        'C': 100,
        'D': 500,
        'M': 1000
    };
    var result = 0,
        length = s.length;
    for (var i = 0; i < length; i++) {
        if (obj[s[i]] >= obj[s[i + 1]] || i == length - 1) {
            result += obj[s[i]];
        } else {
            result -= obj[s[i]];
        }
    }
    return result;
};
console.log(romanToInt('DCXXI'));
